tensorflow/python/estimator/canned/boosted_trees.py

```diff
@@ -192,6 +196,31 @@ def _calculate_num_features(sorted_feature_columns):
   return num_features
 
 
+def _generate_feature_name_mapping(sorted_feature_columns):
```

Comment:
it seems this function needs unit testing. 
we're planning to expand support of feature columns. it will be quite easy to miss this function. Either we should support all feature columns in this function, or error for not supported cases.

Reply:
I think `testFeatureImportancesNamesForCategoricalColumn` verifies the method in part. Do we need to add unit tests for this private method?

Reply:
let's say we added a support for multi dimensional numeric_column. and we forget to update this function. what will happen?

Reply:
Yes, it's a problem if we forget to update this function. 

1. I'm afraid that `_calculate_num_feature` will meet the same problem: if we add new subclass, and we forget to update it. Because we use external function to calculate num_feature and featuer_name, it's easy to forget it when adding new subclass. Perhaps we'd better to make them become the property of FeatureColumn base class, and force every subclass to handle the details. I think the question has beyond the scope of this PR.

2. I think unit tests can only assure old behaviors unchanged. Say, we have supported numeric_column class here, and we use tests to track its behavior. If we add new class, multiple dimensional numeric_column, but we forget to add the corresponding tests. I'm afraid that unit test can do nothing for this case.

Reply:
Hi Yan,
Proposal is erroring out if an unexpected feature column is given. And have unit tests related to do that. So that whenever this method is not updated, we will have a clear error message. 

The other functions you mentioned will cause learning will behave unexpected. So it's likely that we'll caught error in them since there will be unit tests for new columns. But feature importance is not part of learning tests. that's why it's likely to miss. If you assume that it may not be one of you or I or Natalia who will add new features, these kind of errors help new developers. 

please let me know does it make sense or not

Reply:
Oh, my mistake. You're right! The function doesn't support all feature columns, say, multiple dimension numeric_column. 

> Either we should support all feature columns in this function, or error for not supported cases.

I think we should support all feature columns, and add unit tests. How do you think? cc @nataliaponomareva